File: primes/prime.js
    'use strict';
    var min = 2;
    var reason = (function() {
        var m = '';
        return {
            set: function(r) {
                m = r;
            },
            get: function() {
                return m;
            },
            reset: function() {
                m = '';
            },
        };
    })();

    var is = function(n) {
        if (n < min) {
            reason.set('prime must be integer greater than 2');
            return false;
        }

        var sqrt = Math.floor(Math.sqrt(n));
        for (var i = min; i <= sqrt; i++) {
            if (n % i === 0) {
                reason.set(i + ' is divisor of ' + n);
                return false;
            }
        }

        reason.reset();
        return true;
    }

    var first = function(n) {
        var x = min,
            y = 0,
            r = [];
        while (y < n) {
            if (is(x)) {
                r.push(x);
                y++;
            }
            x++;
        }
        return r;
    }

    var between = function(x, y) {
        var r = [];
        for (var i = x; i <= y; i++) {
            if (is(i)) r.push(i);
        }
        return r;
    }

    var why = function() {
        return reason.get();
    }


    exports.is = is;
    exports.first = first;
    exports.between = between;
    exports.why = why;

File: primes/prime.go
package main

import (
	"fmt"
	"math"
)

const Min = 2

type Prime struct{}

func (p Prime) Is(n int) error {
	if n < Min {
		return fmt.Errorf("prime must be integer greater than %d", Min)
	}

	divider := int(math.Floor(math.Sqrt(float64(n))))
	for i := Min; i <= divider; i++ {
		if n%i == 0 {
			return fmt.Errorf("%d is divisor of %d", i, n)
		}
	}

	return nil
}

func (p Prime) First(n int) []int {
	var (
		r         []int = []int{}
		count     int   = 0
		iteration int   = Min
	)
	for count < n {
		if err := p.Is(iteration); err == nil {
			r = append(r, iteration)
			count++
		}
		iteration++
	}
	return r
}

func (p Prime) Between(x, y int) []int {
	var r []int = []int{}
	if x < Min {
		x = Min
	}

	for i := x; i <= y; i++ {
		if err := p.Is(i); err == nil {
			r = append(r, i)
		}
	}
	return r
}
